frontend/src/components/ProductView.tsx

```diff
@@ -0,0 +1,100 @@
+import { useProductViewStyles } from "../styles/ProductViewStyles";
+import { useParams } from "react-router-dom";
+import { Paper, Grid, Button } from "@material-ui/core";
+import { useHistory } from "react-router-dom";
+
+let data = [
+  {
+    name: "product1",
+    category: "category1",
+    price: 15,
+    id: 1,
+    description: "huh",
+    image:
+      "https://sapui5.hana.ondemand.com/test-resources/sap/ui/documentation/sdk/images/HT-1001.jpg",
+  },
+  {
+    name: "product2",
+    category: "category1",
+    price: 30,
+    id: 2,
+    description: "huh",
+    image:
+      "https://sapui5.hana.ondemand.com/test-resources/sap/ui/documentation/sdk/images/HT-1001.jpg",
+  },
+  {
+    name: "product3",
+    category: "category2",
+    price: 10,
+    id: 3,
+    description: "huh",
+    image:
+      "https://sapui5.hana.ondemand.com/test-resources/sap/ui/documentation/sdk/images/HT-1001.jpg",
+  },
+];
+
+interface Product {
+  name: string;
+  category: string;
+  price: number;
+  id: number;
+  image: string;
+  description: string;
+}
+
+function getProduct(id: number): Product | undefined {
+  return data.find((x) => x.id === id);
+}
+
+function ProductDetail() {
+  const classes = useProductViewStyles();
+  const { id } = useParams<{ id?: string | undefined }>();
+  let history = useHistory();
+  let product;
+  if (id !== undefined) {
+    product = getProduct(parseInt(id));
```

Comment:
This getProduct call will be `async` once you will use it to communicate with the backend and you will have to `await` for it and use it in a `useEffect` hook with no dependencies so it will run after the template is rendered